src/execution/operator/persistent/base_csv_reader.cpp

```diff
@@ -135,6 +185,60 @@ bool TryCastTimestampVector(BufferedCSVReaderOptions &options, Vector &input_vec
 	                                                                         count, error_message);
 }
 
+template <class OP, class T>
+bool TemplatedTryCastFloatingVector(BufferedCSVReaderOptions &options, Vector &input_vector, Vector &result_vector,
+                                    idx_t count, string &error_message) {
+	D_ASSERT(input_vector.GetType().id() == LogicalTypeId::VARCHAR);
+	bool all_converted = true;
+	UnaryExecutor::Execute<string_t, T>(input_vector, result_vector, count, [&](string_t input) {
+		T result;
+		if (!OP::Operation(input, result, &error_message)) {
+			all_converted = false;
+		}
+		return result;
+	});
+	return all_converted;
+}
+
+template <class OP, class T>
+bool TemplatedTryCastDecimalVector(BufferedCSVReaderOptions &options, Vector &input_vector, Vector &result_vector,
+                                   idx_t count, string &error_message, uint8_t width, uint8_t scale) {
+	D_ASSERT(input_vector.GetType().id() == LogicalTypeId::VARCHAR);
+	bool all_converted = true;
+	UnaryExecutor::Execute<string_t, T>(input_vector, result_vector, count, [&](string_t input) {
+		T result;
+		if (!OP::Operation(input, result, &error_message, width, scale)) {
+			all_converted = false;
+		}
+		return result;
+	});
+	return all_converted;
+}
+
+template <class T>
+bool TryCastFloatingVector(BufferedCSVReaderOptions &options, Vector &input_vector, Vector &result_vector, idx_t count,
+                           string &error_message, string decimal_separator) {
+	if (decimal_separator == ".") {
+		return TemplatedTryCastFloatingVector<TryCastErrorMessage, T>(options, input_vector, result_vector, count,
+		                                                              error_message);
+	} else {
+		return TemplatedTryCastFloatingVector<TryCastErrorMessageCommaSeparated, T>(
+		    options, input_vector, result_vector, count, error_message);
+	}
+}
+
+template <class T>
+bool TryCastDecimalVector(BufferedCSVReaderOptions &options, Vector &input_vector, Vector &result_vector, idx_t count,
+                          string &error_message, uint8_t width, uint8_t scale, string decimal_separator) {
+	if (decimal_separator == ".") {
+		return TemplatedTryCastDecimalVector<TryCastToDecimal, T>(options, input_vector, result_vector, count,
```

Comment:
If we can ensure that the decimal separator is `,` at a higher level (since we can use the standard cast for a decimal separator of `.`) we can remove this if statement and simplify the code